src/unix.rs

```diff
@@ -21,13 +21,27 @@ pub struct Acquired {
 }
 
 impl Client {
-    pub fn new(limit: usize) -> io::Result<Client> {
+    pub fn new(mut limit: usize) -> io::Result<Client> {
+        assert!(limit <= 4096, "Can't have limit larger than 4096 since the pipe would block forever when writing to it");
+
         let client = unsafe { Client::mk()? };
+
         // I don't think the character written here matters, but I could be
         // wrong!
-        for _ in 0..limit {
-            (&client.write).write_all(&[b'|'])?;
+        //
+        // Also, it is highly unlikely to have a machine with more than
+        // 128 cores.
```

Comment:
I think it's ok to remove this comment since nowadays I'm actually working on machines with ~100 cores so this "highly unlikely" doesn't seem like it'll be true for very long.

Reply:
Removed!